xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp

```diff
@@ -766,3 +766,18 @@ DEBUG_INFO_VIDEO CRendererBase::GetDebugInfo(int idx)
 
   return info;
 }
+
+bool CRendererBase::RenderAsHDR(const AVColorPrimaries primaries,
```

Comment:
```suggestion
bool CRendererBase::CanRenderAsHDR(const AVColorPrimaries primaries,
```

Reply:
With a different name if you don't mind (IntendToRenderAsHDR?). This is to communicate the intention of the rendering as SDR or HDR, because this will be called by GetWeight and Configure have to know before rendering actually starts, before refresh rate and/or Windows HDR switch.
It was not optimal to call the function from CheckVideoParameters, which should be fed with the actual mode of rendering as set by RendererBase (m_HdrType) > now done through new accessor ActualRenderAsHDR()
Although not well handled yet, it's possible the renderer may be unable to switch to the desired mode and there could be a difference between intention and actual. As a benefit the HDR switch enabled parameter is not needed anymore.

The related logic in RendererBase could be consolidated, but that wasn't the purpose of this PR, maybe later.

Reply:
to be honest I don't remember all flow of the preparation of rendering now so I have no opinion at this. Just try